warehouse/internal/repo/schema.go

```diff
@@ -252,3 +253,20 @@ func (sh *WHSchema) GetTablesForConnection(ctx context.Context, connections []wa
 
 	return tables, nil
 }
+
+func (sh *WHSchema) SetExpiryForDestination(ctx context.Context, destinationID string, expiresAt time.Time) (int, error) {
+	query := `
+		UPDATE ` + whSchemaTableName + `
+		SET expires_at = $1, updated_at = $2
+		WHERE destination_id = $3
+	`
+	result, err := sh.db.ExecContext(ctx, query, expiresAt, sh.now(), destinationID)
+	if err != nil {
+		return 0, fmt.Errorf("updating expiry: %w", err)
+	}
+	rowsAffected, err := result.RowsAffected()
+	if err != nil {
+		return 0, fmt.Errorf("getting rows affected: %w", err)
+	}
+	return int(rowsAffected), nil
+}
```

Comment:
affected rows count is not being used, Can't we do something like this?
```suggestion
func (sh *WHSchema) SetExpiryForDestination(ctx context.Context, destinationID string, expiresAt time.Time) error {
	query := `
		UPDATE ` + whSchemaTableName + `
		SET expires_at = $1, updated_at = $2
		WHERE destination_id = $3
	`
	result, err := sh.db.ExecContext(ctx, query, expiresAt, sh.now(), destinationID)
	if err != nil {
		return fmt.Errorf("updating expiry: %w", err)
	}
	rowsAffected, err := result.RowsAffected()
	if err != nil {
		return fmt.Errorf("getting rows affected: %w", err)
	}
	if rowsAffected == 0 {
		return fmt.Errorf("no rows affected")
	}
	return nil
}
```

Reply:
I don't think we should be returning an error if `rowsAffected == 0`. It is a valid case where there are no entries for a destination in the table and user triggers the API. If you want, I can get rid of `rowsAffected` completely

Reply:
Yeah, let's get rid of it if not being used.

Reply:
Removed